applications/nrf_desktop/src/modules/ble_qos.c

```diff
@@ -0,0 +1,722 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <device.h>
+#include <drivers/uart.h>
+#include <misc/byteorder.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+
+#include "ble_controller_hci_vs.h"
+
+#include "chmap_filter.h"
+
+#define MODULE ble_qos
+#include "module_state_event.h"
+#include "ble_event.h"
+#include "config_event.h"
+#include "hid_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_BLE_QOS_LOG_LEVEL);
+
+#define INVALID_BLACKLIST 0xFFFF
+
+#if CONFIG_DESKTOP_BLE_QOS_STATS_PRINTOUT_ENABLE
+# if DT_NORDIC_NRF_USBD_USBD_0_NUM_IN_ENDPOINTS < 4
+# error Too few USB IN Endpoints enabled. \
+	Modify appropriate dts.overlay to increase num-in-endpoints to 4 or more
+# endif
+#endif
+
+#ifndef ROUNDED_DIV
+#define ROUNDED_DIV(a, b)  (((a) + ((b) / 2)) / (b))
+#endif /* ROUNDED_DIV */
+
+#if CONFIG_DESKTOP_BLE_QOS_STATS_PRINTOUT_ENABLE
+#define THREAD_STACK_SIZE		     \
+	(CONFIG_DESKTOP_BLE_QOS_STACK_SIZE + \
+	 CONFIG_DESKTOP_BLE_QOS_STATS_PRINT_STACK_SIZE)
+#else
+#define THREAD_STACK_SIZE CONFIG_DESKTOP_BLE_QOS_STACK_SIZE
+#endif
+
+#define THREAD_PRIORITY K_PRIO_PREEMPT(K_LOWEST_APPLICATION_THREAD_PRIO)
```

Comment:
`K_PRIO_PREEMPT` is not needed. If `CONFIG_PREEMPT_ENABLED` is enabled (and at least 2) `K_LOWEST_APPLICATION_THREAD_PRIO` will be preemptable.

This said K_PRIO_PREEMPT does not change much...

Reply:
The readability improves somewhat. Without K_PRIO_PREEMPT the reader must investigate K_LOWEST_APPLICATION_THREAD_PRIO more closely